### PR TITLE
refactor(core): move recovery code to `utxo/recovery/`

### PR DESCRIPTION
**Pull request by OttoAllmendinger**
The broad topic of this PR is moving utxo recovery code that is scattered around the source tree and inheritance hierarchy into a central location `utxo/recovery`. 

The code that I moved around is largely unchanged. I removed `Bluebird` wrappers and simplified the construction interface for external APIs to be less reliant on the global environment.

I checked the usage for all methods that I removed from `AbstractCoin` / `AbstractUtxoCoin` in https://github.com/BitGo/wallet-recovery-wizard

Issue: BG-38773

### CODE REVIEW

File: modules/core/src/v2/coins/utxo/recovery/blockchairApi.ts [hunk 1]
```diff
@@ -1,56 +1,63 @@
...
+        break;
+      case 'zec':
+        blockchain = 'zcash';
+        break;
+      case 'tbsv':
+      case 'tbch':
+      case 'tbcha':
+        // FIXME: these only exist to satisfy tests
+        blockchain = 'mock-' + coinName;
+        break;
```
Comment (OttoAllmendinger) [inline modules/core/src/v2/coins/utxo/recovery/blockchairApi.ts:43]: note that this was also broken prior to this change: the code just assumed that the testnet url always is `${blockchain}/testnet` when it only is for bitcoin

File: modules/core/src/v2/coins/utxo/recovery/blockchairApi.ts [hunk 2]
```diff
@@ -1,56 +1,63 @@
...
-    this.apiToken = apiToken;
+    return new BlockchairApi(`https://api.blockchair.com/${blockchain}`, apiToken);
   }
 
-  static getBaseUrl(env: string, coin: string): string {
-    let url;
-    if (mainnetBase.includes(env)) {
-      url = `https://api.blockchair.com/${coin}`;
-    } else if (testnetBase.includes(env) || devBase.includes((env))) {
-      url = `https://api.blockchair.com/${coin}/testnet`;
```
Comment (OttoAllmendinger) [inline modules/core/src/v2/coins/utxo/recovery/blockchairApi.ts:36]: as stated before: only correct for `coin === 'bitcoin'`

File: modules/core/src/v2/coins/utxo/recovery/insightApi.ts [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    switch (coinName) {
+      case 'dash':
+        return 'https://insight.dash.org/insight-api';
+      case 'tdash':
+        return 'https://testnet-insight.dashevo.org/insight-api';
+      case 'ltc':
+        return 'https://insight.litecore.io/api';
+      case 'tltc':
+        // FIXME: defunct
+        return 'https://insight.litecore-test.io/api';
```
Comment (OttoAllmendinger) [inline modules/core/src/v2/coins/utxo/recovery/insightApi.ts:23]: keeping this for so tests are happy

File: modules/core/src/v2/coins/utxo/recovery/backupKeyRecovery.ts [hunk 4]
```diff
@@ -101,6 +104,82 @@ function formatForOfflineVault(
...
+ * This function is only intended for non-bitgo recovery transactions, when it is necessary
+ * to calculate the rough fee needed to pay to Keyternal. We are okay with approximating,
+ * because the resulting price of this function only has less than 1 dollar influence on the
+ * fee that needs to be paid to Keyternal.
+ *
+ * See calculateFeeAmount function:  return Math.round(feeAmountUsd / currentPrice * self.getBaseFactor());
+ *
+ * This end function should not be used as an accurate endpoint, since some coins' prices are missing from the provider
+ */
+async function getRecoveryMarketPrice(coin: AbstractUtxoCoin): Promise<string> {
```
Comment (OttoAllmendinger) [inline modules/core/src/v2/coins/utxo/recovery/backupKeyRecovery.ts:119]: could also go to `coinGeckoApi.ts` in some future PR

**Review by sangaman — APPROVED** [review]
Looks like good restructuring to me.

**Review by tylerlevine — APPROVED** [review]
outstanding clean up, this is great work 🚀

**Pull request merged**